lib/product_details.dart

```diff
@@ -94,7 +95,9 @@ class _ProductDetailsState extends State<ProductDetails> {
                       ),
                     ),
                     ElevatedButton(
-                      onPressed: () {},
+                      onPressed: () {
+                        cart.addItem(details!);
```

Comment:
you could show snack bar after adding the item, to notify the user that the product is added.